File: extension/src/test/java/com/wesleyhome/test/jupiter/java/provider/EnumParameterDataProviderTest.java
package com.wesleyhome.test.jupiter.java.provider;

import com.wesleyhome.test.jupiter.provider.EnumParameterDataProvider;
import com.wesleyhome.test.jupiter.provider.TestParameter;
import java.util.List;

public class EnumParameterDataProviderTest extends ParameterDataProviderTest<EnumParameterDataProvider, Enum<?>> {

    @Override
    protected Class<Enum<?>> initializeParameterType() {
        return null;
    }

    @Override
    protected TestParameter createTestParameter(boolean isNullable) {
        return new TestParameter("testCreateParameterOptionsData", kotlin(StateValueJava.class), isNullable, List.of());
    }

    @Override
    protected TestParameter createFalseProvidesForTestParameter() {
        return new TestParameter("testCreateParameterOptionsData", kotlin(Integer.class), true, List.of());
    }

}

enum StateValueJava {
    PENDING_JAVA,
    STARTED_JAVA,
    DONE_JAVA
}

File: extension/src/test/kotlin/com/wesleyhome/test/jupiter/provider/EnumParameterDataProviderTest.kt
package com.wesleyhome.test.jupiter.provider

import kotlin.reflect.KClass

@Suppress("UNCHECKED_CAST")
internal class EnumParameterDataProviderTest : ParameterDataProviderTest<EnumParameterDataProvider, Enum<*>>() {

    override val parameterType: KClass<Enum<*>> by lazy { StateValue::class as KClass<Enum<*>> }

    override fun createFalseProvidesForTestParameter(): TestParameter {
        return createTestParameter(true).copy(type = Int::class)
    }
}

enum class StateValue {
    PENDING,
    STARTED,
    DONE
}
